osidb/models.py

```diff
@@ -2919,6 +2933,45 @@ def validate(self, *args, **kwargs):
         super().clean_fields(*args, **kwargs)
 
 
+class Package(models.Model):
+    """Model representing a package with connected versions."""
+
+    uuid = models.UUIDField(primary_key=True, default=uuid.uuid4, editable=False)
+
+    flaw = models.ForeignKey(
+        Flaw,
+        related_name="package_versions",
+        on_delete=models.CASCADE,
+    )
+
+    package = models.CharField(max_length=2058)
+
+    # TODO: This is temporarily present to make existing tests happy. Reconsider whether to delete.
+    def validate(self, *args, **kwargs):
+        super().clean_fields(*args, **kwargs)
+
+
+class PackageVer(models.Model):
+    """Model representing a package version"""
+
+    uuid = models.UUIDField(primary_key=True, default=uuid.uuid4, editable=False)
+
+    package = models.ForeignKey(
+        Package,
+        related_name="versions",
+        on_delete=models.CASCADE,
+    )
+
+    class Meta:
+        verbose_name = "Version"
+
+    version = models.CharField(max_length=1024)
+
+    # TODO: This is temporarily present to make existing tests happy. Reconsider whether to delete.
+    def validate(self, *args, **kwargs):
+        super().clean_fields(*args, **kwargs)
```

Comment:
`validate` or `version`?

Reply:
I think he thought the whole `PackageVer` model.

Reply:
I do not see a reason for two models for a package and its version. I mean, the package without the version seems to be incomplete. I think this is a bad design.

Reply:
`PackageVer` exists so that serialization into the CVE 5.0-compatible structure is straightforward. Having ArrayField and the current API structure would require custom de/serializers and bending/duplicating the custom OSIDB mixins for serializers. If we end up deciding to drop CVE 5.0 compatibility, we can use ArrayField with only 1 model, just like the other models do.

Reply:
Closed as per https://github.com/RedHatProductSecurity/osidb/pull/307#issuecomment-1713798433